coordinator/storage/monitor.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2014 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/zenoss/glog"
+)
+
+const monitorSubDir string = "/monitor"
+
+var (
+	monitoredHosts     map[string]bool
+	monitoredHostsLock sync.RWMutex
+)
+
+func init() {
+	monitoredHosts = make(map[string]bool)
+}
+
+// UpdateRemoteMonitorFile writes a tiny file to the DFS volume at given cycle
+func UpdateRemoteMonitorFile(localPath string, writeInterval time.Duration, ipAddr string) {
+	glog.Infof("writing remote monitor file to DFS volume %s at interval: %s", localPath, writeInterval)
+
+	for {
+		remoteFile := path.Join(localPath, monitorSubDir, ipAddr)
+		glog.V(0).Infof("writing DFS file %s", remoteFile)
+		if err := ioutil.WriteFile(remoteFile, []byte(ipAddr), 0600); err != nil {
+			glog.Warningf("unable to write DFS file %s: %s", remoteFile, err)
+		}
+
+		// wait for next cycle or shutdown
+		select {
+		case <-time.After(writeInterval):
+		}
+	}
+}
+
+//
+func AddMonitorRemoteHost(ipAddr string) {
+	monitoredHostsLock.Lock()
+	defer monitoredHostsLock.Unlock()
+	monitoredHosts[ipAddr] = true
+}
+
+//
+func SetMonitorRemoteHost(ipAddr string, hasUpdated bool) {
```

Comment:
I can't see anywhere where we actually use the bool part of an entry in this map - am I missing something?